rfcs/0022-s3-source.md

```diff
@@ -0,0 +1,75 @@
+---
+feature: S3 Source with SQS
+authors:
+  - "Bohan Zhang"
+start_date: "2022/11/18"
+---
+
+# S3 Source with SQS
+
+## Motivation
+
+Support load data from Amazon S3 and other Simple Storage Services.
+
+## Background
+
+The previous two versions RFCs are available as follows:
+
+* [Version 1](https://www.notion.so/risingwave-labs/Deprecated-RFC-S3-Source-Design-632aa78264b347f9a4a37fd89a91d785): presents the basic abstraction of treating a file as a partition, but here we ignore the fact that a source can be relied on by multiple MVs, while sqs can only be consumed once. This results in inconsistent data across multiple MVs.
+* [Version 2](https://github.com/risingwavelabs/rfcs/pull/19): want the executors to list s3 buckets independently and maintain their own state. This solution is not accepted because listing the bucket periodically is too expensive ([price policy](https://aws.amazon.com/s3/pricing/)) and we want to use SQS to describe the incremental data.
```

Comment:
The combinition of S3 + SQS is very rare in my (limited) experience. I have never seen any system supported for it. Even Airbyte only supports the `ls` approach (https://airbytehq.github.io/integrations/sources/s3/).

My concern is that in most cases, users may not have full control over his S3 buckets, thus, he is unable or unwilling to turn on SQS for RisingWave. I suppose this could be an optional feature and more than 2/3 users don't use it.

Personally I recommend to focus on S3 **without** SQS at this time.

Reply:
cc. @neverchanje @tabVersion

Reply:
If we agree to give up SQS support, the design can also be simplified a lot:

- The `Chain` operator between `List` and `Fetch` is not needed anymore
- The `List` stage can be completely stateless

Actually, I guess the whole design can be refactored...